### PR TITLE
feat(web): project recycle bin

### PR DESCRIPTION
**Pull request by mkumbobeaty**
# Overview
- Add delete functionality in project dashboard and project setting
- Implement Recycle bin page
## What I haven't done
- disable project not to be open in other tabs after being moved to recycle bin.
## Release Notes

- **New Features**
  - Introduced project deletion functionality, including a new modal for confirming deletions.
  - Added support for managing deleted projects and a recycle bin interface for recovery actions.
  - New components for project removal and recycle bin management, including `ProjectRemoveModal` and `ProjectDeleteModal`.
  - Added a new `NotFound` component for improved user experience when a project is not found.

- **Improvements**
  - Enhanced project management tests for better reliability.
  - Updated project data structures to include deletion status, improving clarity and functionality.
  - Updated the `ProjectGridViewItem` and `ProjectListViewItem` components to handle project deletions and integrate with the recycle bin.
  - Improved visibility and management of deleted projects through new hooks and API enhancements.
  - Streamlined the project removal process with updated messaging and modal integration.

- **Translations**
  - Updated English and Japanese translation files with new keys for project management actions, enhancing clarity in user messaging regarding project deletion and recovery processes.

### CODE REVIEW

File: web/src/services/api/projectApi.ts [hunk 1]
```diff
@@ -106,6 +107,20 @@ export default () => {
...
+    });
+
+    const deletedProjects = useMemo(
+      () => data?.deletedProjects.nodes,
+      [data?.deletedProjects]
+    );
+
+    return { deletedProjects, ...rest };
+  }, []);
+
```
[the remark this answers is not in the document]
Comment (mkumbobeaty) [reply web/src/services/api/projectApi.ts:123]: it is correct we need string not object

File: web/src/beta/features/Dashboard/ContentsContainer/Projects/ProjectDeleteModal.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          <Typography size="body">
+            {t(
+              "This means the project will no longer be published. But you can still see and restore you project from recycle bin."
+            )}
+          </Typography>
+        </Wrapper>
+      </ModalPanel>
+    </Modal>
+  );
+};
```
[the remark this answers is not in the document]
Comment (mkumbobeaty) [reply web/src/beta/features/Dashboard/ContentsContainer/Projects/ProjectDeleteModal.tsx:53]: this already handled in other file

File: web/src/beta/features/Dashboard/ContentsContainer/RecycleBin/index.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                      )}
+                    </>
+                  ))}
+                </ProjectsGroup>
+              </ProjectsContainer>
+              {isLoading && (
+                <LoadingWrapper>
+                  <Loading relative />
+                </LoadingWrapper>
+              )}
```
[the remark this answers is not in the document]
Comment (mkumbobeaty) [reply web/src/beta/features/Dashboard/ContentsContainer/RecycleBin/index.tsx:60]: No need to load when no project

File: web/src/beta/features/Dashboard/ContentsContainer/RecycleBin/index.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  "@media (max-width: 1200px)": {
+    gridTemplateColumns: "repeat(3, 1fr)"
+  },
+  "@media (max-width: 900px)": {
+    gridTemplateColumns: "repeat(2, 1fr)"
+  },
+  "@media (max-width: 600px)": {
+    gridTemplateColumns: "1fr"
+  }
+}));
```
[the remark this answers is not in the document]
Comment (mkumbobeaty) [reply web/src/beta/features/Dashboard/ContentsContainer/RecycleBin/index.tsx:98]: we dont have theme.breakpoints.down in ouw system

File: web/src/beta/features/Dashboard/ContentsContainer/RecycleBin/hooks.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      const updatedProject = {
+        projectId: project.id,
+        deleted: false
+      };
+
+      await useUpdateProjectRemove(updatedProject);
+    },
+
+    [useUpdateProjectRemove]
+  );
```
[the remark this answers is not in the document]
Comment (mkumbobeaty) [reply web/src/beta/features/Dashboard/ContentsContainer/RecycleBin/hooks.ts:43]: this is already handled in useUpdateProjectRemove

**Review by pyshx — APPROVED** [review]

**Review by airslice — APPROVED** [review]

**Pull request merged**